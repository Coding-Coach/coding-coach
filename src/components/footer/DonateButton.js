import React from 'react';
import { translate } from 'react-i18next';
import 'i18n/i18n';
import Config from 'config/constants';
import Image from 'components/image/Image';

function DonateButton({ t }) {
  return (
    <form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
      <input type="hidden" name="cmd" value="_s-xclick" />
      <input type="hidden" name="hosted_button_id" value={Config.payments.paypal} />
      <input
        type="image"
        src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif"
        border="0"
        name="submit"
        alt={t('footer-donate-paypal')}
      />
      <Image alt={t('footer-donate')} src="https://www.paypalobjects.com/de_DE/i/scr/pixel.gif" />
    </form>
  );
}

export default translate('translations')(DonateButton);
