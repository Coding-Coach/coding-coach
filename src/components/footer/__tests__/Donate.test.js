import React from 'react';
import { render } from 'react-testing-library';
import DonateButton from '../DonateButton';

describe('components/footer/DonateButton', () => {
  it('should render the donate button', () => {
    const { container } = render(<DonateButton />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
