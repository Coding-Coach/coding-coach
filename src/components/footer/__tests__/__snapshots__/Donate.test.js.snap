// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/footer/DonateButton should render the donate button 1`] = `
<form
  action="https://www.paypal.com/cgi-bin/webscr"
  method="post"
  target="_top"
>
  <input
    type="hidden"
    name="cmd"
    value="_s-xclick"
  />
  <input
    type="hidden"
    name="hosted_button_id"
    value="123abc"
  />
  <input
    type="image"
    src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif"
    border="0"
    name="submit"
    alt="PayPal - The safer, easier way to pay online!"
  />
  <img
    src="https://www.paypalobjects.com/de_DE/i/scr/pixel.gif"
    alt="Donate!"
  />
</form>
`;
